sync_telegram/models/sync_project.py

```diff
@@ -37,22 +47,104 @@ def _eval_context_telegram(self, secrets, eval_context):
+            return html[5:-6]
+
+        def _url2base64(url):
+            r = requests.get(url)
+            datas = base64.b64encode(r.content)
+            return datas
+
+        def create_mail_сhannel(partners):
+            return (
+                self.env["mail.channel"]
+                .sudo()
+                .create(
+                    {
+                        "channel_partner_ids": [
+                            (4, partner_id) for partner_id in partners
+                        ],
+                        "public": "private",
+                        "channel_type": "telegram",
+                        "email_send": False,
+                        "name": ", ".join(
+                            self.env["res.partner"]
+                            .sudo()
+                            .browse(partners)
+                            .mapped("name")
+                        ),
+                    }
+                )
             )
-            bot.sendMessage(chat_id, *args, **kwargs)
+
+        def sendMessage(chat_id, html, *args, **kwargs):
+            log_transmission("%s@telegram" % chat_id, "Message: %s" % html)
+            bot.sendMessage(chat_id, _html_sanitize_telegram(html), *args, **kwargs)
+
+        def sendPhoto(chat_id, datas):
+            log_transmission("%s@telegram" % chat_id, "Photo sent")
+            bot.sendPhoto(chat_id, photo=base64.b64decode(datas))
+
+        def sendDocument(chat_id, name, datas):
+            log_transmission("%s@telegram" % chat_id, "Document sent")
+            bot.sendDocument(chat_id, filename=name, document=base64.b64decode(datas))
+
+        def getUserPhoto(chat_id):
+            photo_list = bot.getUserProfilePhotos(chat_id).photos
+            if not photo_list:
+                return None
+            else:
+                photo_path = bot.getFile(photo_list[0][0]["file_id"]).file_path
+                log(
+                    "The user's photo has been added to the partner record.",
+                    name="%s@telegram" % chat_id,
+                )
+                return _url2base64(photo_path)
+
+        def getFileBinary(chat_id, file_type, file_data):
+            try:
+                if file_type == "document":
+                    file_name = file_data.file_name
+                else:
+                    file_name = bot.getFile(file_data).file_path.split("/")[-1]
+
+                content = bot.getFile(file_data).download_as_bytearray()
+
+                log(
+                    "Attachment from telegram received: file_name: %s" % file_name,
+                    name="%s@telegram" % chat_id,
+                )
+                return [file_name, content]
+
+            except error.BadRequest as err:
+                log(
+                    "Error processing attachment from telegram: %s" % err,
+                    name="%s@telegram" % chat_id,
+                    level=LOG_ERROR,
+                )
+                return ["error", err]
```

Comment:
А что если файл называется "error" ?
```suggestion
                raise
```